### PR TITLE
feat: Improve lexer error recovery

### PR DESCRIPTION
**Pull request by max-sixty**
Upgrade to Chumsky 0.9.2 and add back recovery

### CODE REVIEW

File: prql-compiler/src/test.rs [hunk 1]
```diff
@@ -2954,6 +2954,20 @@ fn test_errors() {
...
+     1 │ Mississippi has four S’s and four I’s.
+       ·                                    ┬
+       ·                                    ╰── unexpected ’
+    ───╯
+    Error:
+       ╭─[:1:39]
+       │
+     1 │ Mississippi has four S’s and four I’s.
+       ·                                       ┬
+       ·                                       ╰── identifier expected an identifier or *, but found
```
Comment (max-sixty) [inline prql-compiler/src/test.rs]: It's not great prose, but we can iterate...

**Comment by max-sixty** [comment]
Unfortunately it has volatile outputs...

Is this expected? Can we make error recovery consistent?

**Comment by aljazerzen** [comment]
What do you mean "volatile"? Non deterministic, so different every run?

I'm guessing you are talking about order of expected tokens. We can sort that before displaying.

File: prql-compiler/src/test.rs (hunk 1 above)
Comment (aljazerzen) [reply prql-compiler/src/test.rs]: I did not know that at the time, but expected token `None` means end of file. In the convert_parser_error I'm converting that `None` into empty string, which causes this output.

**Comment by max-sixty** [comment]
Yes exactly, non-deterministic is a better term.

Ok, great if we construct the message ourselves, easy for us to sort the items, I can have a look

**Comment by aljazerzen** [comment]
https://github.com/PRQL/prql/blob/main/prql-compiler/src/parser/mod.rs#L69-L116

File: prql-compiler/src/test.rs (hunk 1 above)
Comment (max-sixty) [reply prql-compiler/src/test.rs]: Great, fixed in #2008, thanks

**Pull request merged**